images/win/scripts/Installers/Install-JavaTools.ps1

```diff
@@ -87,19 +96,37 @@ function Install-JavaFromAdoptOpenJDK {
     New-Item -ItemType File -Path $javaVersionPath -Name "$Architecture.complete" | Out-Null
 }
 
-$jdkVersions = (Get-ToolsetContent).java.versions
+$jdkVendors = (Get-ToolsetContent).java.vendors.name
+$defaultVendor = (Get-ToolsetContent).java.default_vendor
 $defaultVersion = (Get-ToolsetContent).java.default
 
-foreach ($jdkVersion in $jdkVersions) {
-    Install-JavaFromAdoptOpenJDK -JDKVersion $jdkVersion
+# We get the difference between Adopt and Adoptium and then assign it as Adopt's JAVA_HOME
+$uniqVersion = (Get-ToolsetContent).java.vendors.versions | Group-Object -noelement | Where-Object {$_.count -eq 1} | Select-Object -expand Name
+
+foreach ($jdkVendor in $jdkVendors) {
+
+    $jdkVersionsToInstall = ((Get-ToolsetContent).java.vendors | Where-Object { $_.name -eq $jdkVendor }).versions
+    $isDefaultVendor = $jdkVendor -eq $defaultVendor
+
+    foreach ($jdkVersionToInstall in $jdkVersionsToInstall)
+    {
+        Install-JavaJDK -FullVendorName $jdkVendor -JDKVersion $jdkVersionToInstall
+
+        if ($isDefaultVendor) {
+            Set-JavaPath -Version $jdkVersionToInstall -DefaultVersion $defaultVersion -FullVendorName $jdkVendor
+
+        }
 
-    if ($jdkVersion -eq $defaultVersion) {
-        Set-JavaPath -Version $jdkVersion -Default
-    } else {
-        Set-JavaPath -Version $jdkVersion
     }
+
 }
 
+# Here we setup JAVA_HOME_13_X64 as this is the only Adopt version we want
+# Adoptium has only 8 & 11 so we do not have to check vendor name explicily as we know
+# 13 belongs to Adopt only.
+
+Set-JavaPath -Version $uniqVersion -FullVendorName "Adopt"
```

Comment:
May be just hardcode Java 13 here?
```
Set-JavaPath -Version "13" -FullVendorName "Adopt"
```

I don't see much profit of calculating `uniqVersion` in runtime because this logic will fail anyway if we remove Java 13 or add new Java from Adopt

Reply:
Hardcoded